app/models/mixins/process_tasks_mixin.rb

```diff
@@ -88,11 +84,51 @@ def invoke_tasks_remote(options)
       end
     end
 
-    # This should be overridden by any class including this module
-    # and expecting to be asked to invoke tasks for a remote region
-    def invoke_api_tasks(_hostname, _remote_options)
-      raise NotImplementedError
+    # Override as needed to handle differences between API actions and method names
+    def action_for_task(task)
+      task
+    end
+
+    def invoke_api_tasks(api_client, remote_options)
+      collection_name = Api::CollectionConfig.new.name_for_klass(self)
+      unless collection_name
+        _log.error("No API entpoint found for class #{name}")
+        raise NotImplementedError
+      end
+
+      collection   = api_client.send(collection_name)
+      action       = action_for_task(remote_options[:task])
+      post_args    = remote_options[:args] || {}
+      resource_ids = remote_options[:ids]
+
+      if resource_ids.present?
+        resource_ids.each do |id|
+          obj = collection.find(id)
+          _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
+          obj.send(action, post_args)
+        end
```

Comment:
are there a lot of resource_ids usually in this calling path ?  If so, you could do a single bulk request on the collection.